internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -149,15 +155,18 @@ type ASC struct {
 	conditions *[]conditionsv1.Condition
 
 	/* properties. use this field for cross cluster communication */
-	properties map[string]interface{}
+	properties *map[string]interface{}
```

Comment:
The zero-value of a map is `nil` and they are generally treated as references. Why do you need a pointer here?

Here's an example if it helps: https://go.dev/play/p/l5YfudQhWT1

Reply:
Nice, wasn't sure about it. Fixed.